### DISCUSSION TITLE
Modifying Artifact and Log Retention Settings via the GitHub REST API

### DISCUSSION THREAD

**Discussion by syakovenko81** [community/community#114381] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hello everyone,

I'm trying to modify the "Artifact and log retention" settings for a specific repository using the GitHub REST API. 
While I've found instructions for doing this through the GitHub UI [[link](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#setting-the-retention-period-for-a-repository)], I'm unable to find similar guidance for accomplishing it via the api call.

If anyone has experience with this or knows of a workaround, I'd greatly appreciate your insights.

Thank you!

**Comment by vosscodes** [comment]
I am shocked that this apparently isn't in the API. Organization settings imply you are setting a _default_, but it is actually a hard cap _and_ the organization wide default setting.

Needing 4-5 repos at >1 year log retention when you have hundreds of repos in your org is brutal if this can't be done via the API.
